crates/bevy_core_pipeline/src/core_2d/main_transparent_pass_2d_node.rs

```diff
@@ -3,67 +3,50 @@ use bevy_ecs::prelude::*;
 use bevy_render::{
     camera::ExtractedCamera,
     diagnostic::RecordDiagnostics,
-    render_graph::{Node, NodeRunError, RenderGraphContext},
+    render_graph::{NodeRunError, RenderGraphContext, ViewNode},
     render_phase::SortedRenderPhase,
     render_resource::RenderPassDescriptor,
     renderer::RenderContext,
-    view::{ExtractedView, ViewTarget},
+    view::ViewTarget,
 };
 #[cfg(feature = "trace")]
 use bevy_utils::tracing::info_span;
 
-pub struct MainPass2dNode {
-    query: QueryState<
-        (
-            &'static ExtractedCamera,
-            &'static SortedRenderPhase<Transparent2d>,
-            &'static ViewTarget,
-        ),
-        With<ExtractedView>,
-    >,
-}
-
-impl FromWorld for MainPass2dNode {
-    fn from_world(world: &mut World) -> Self {
-        Self {
-            query: world.query_filtered(),
-        }
-    }
-}
+#[derive(Default)]
+pub struct MainTransparentPass2dNode {}
 
-impl Node for MainPass2dNode {
-    fn update(&mut self, world: &mut World) {
-        self.query.update_archetypes(world);
-    }
+impl ViewNode for MainTransparentPass2dNode {
+    type ViewQuery = (
+        &'static ExtractedCamera,
+        &'static SortedRenderPhase<Transparent2d>,
+        &'static ViewTarget,
+    );
 
-    fn run(
+    fn run<'w>(
         &self,
         graph: &mut RenderGraphContext,
-        render_context: &mut RenderContext,
-        world: &World,
+        render_context: &mut RenderContext<'w>,
+        (camera, transparent_phase, target): bevy_ecs::query::QueryItem<'w, Self::ViewQuery>,
+        world: &'w World,
     ) -> Result<(), NodeRunError> {
         let view_entity = graph.view_entity();
-        let Ok((camera, transparent_phase, target)) = self.query.get_manual(world, view_entity)
-        else {
-            // no target
-            return Ok(());
-        };
 
-        {
+        if !transparent_phase.items.is_empty() {
             #[cfg(feature = "trace")]
-            let _main_pass_2d = info_span!("main_pass_2d").entered();
+            let _main_pass_2d = info_span!("main_transparent_pass_2d").entered();
 
             let diagnostics = render_context.diagnostic_recorder();
 
             let mut render_pass = render_context.begin_tracked_render_pass(RenderPassDescriptor {
-                label: Some("main_pass_2d"),
+                label: Some("main_transparent_pass_2d"),
                 color_attachments: &[Some(target.get_color_attachment())],
+                // TODO 2d depth buffer
```

Comment:
I think this TODO either needs to be more specific or be removed

Reply:
Ah, I see it will be removed in the next PR

Reply:
I could probably remove it from this PR, but yeah it will be removed in the follow up PR. It was useful when I was still working on actually adding the 2d depth buffer.

Reply:
I removed the TODO